src/alluvium/avro.py

```diff
@@ -0,0 +1,251 @@
+"""
+Defines the schema you expect from your CDC events.
+
+Provides synctactic sugar for Any types and Either Types.
+"""
+
+from abc import ABC, abstractmethod
+from enum import Enum
+from typing import Any, List, Optional, Union
+
+
+class InvalidSchemaException(Exception):
+    def __init__(self, field_type, message, name=None):
+        self.error_field_type = field_type
+        self.name = name
+        super(InvalidSchemaException, self).__init__(
+            self, "Error with field <{} {}>. {}".format(field_type, name, message)
+        )
+
+
+class AvroPrimitiveType(Enum):
+    NULL = "null"
+    BOOLEAN = "boolean"
+    INTEGER = "int"
+    LONG = "long"
+    FLOAT = "float"
+    DOUBLE = "double"
+    BYTES = "bytes"
+    STRING = "string"
+
+
+class AvroComplexType(Enum):
+    RECORD = "record"
+    ENUM = "enum"
+    ARRAY = "array"
+    MAP = "map"
+    UNIONS = "unions"
+    FIXED = "fixed"
+
+
+class AvroCustomType(Enum):
+    ANY = "any"
+
+
+def resolve_primitive_type(avro_type: Union[AvroPrimitiveType, AvroCustomType]):
```

Comment:
Unimportant code style comment: consider moving this function to after class FieldOrdering to demarcate the transition from type definition to class definition

Reply:
Good call! However, what if we instead moved Field Ordering to record field because field ordering is actually a field concept and not a type or a schema concept.

Reply:
That makes more sense